src/wrap-request.ts

```diff
@@ -1,17 +1,31 @@
+import type { EndpointDefaults, OctokitResponse } from "@octokit/types";
+import type { State } from "./types";
+
 const noop = () => Promise.resolve();
 
-// @ts-expect-error
-export function wrapRequest(state, request, options) {
+export function wrapRequest(
+  state: State,
+  request: (
+    options: Required<EndpointDefaults>,
+  ) => OctokitResponse<any> | Promise<OctokitResponse<any>>,
+  options: Required<EndpointDefaults>,
+) {
   return state.retryLimiter.schedule(doRequest, state, request, options);
 }
 
-// @ts-expect-error
-async function doRequest(state, request, options) {
+async function doRequest(
+  state: State,
+  request: (
+    options: Required<EndpointDefaults>,
+  ) => OctokitResponse<any> | Promise<OctokitResponse<any>>,
+  options: Required<EndpointDefaults>,
+) {
   const isWrite = options.method !== "GET" && options.method !== "HEAD";
   const { pathname } = new URL(options.url, "http://github.test");
   const isSearch = options.method === "GET" && pathname.startsWith("/search/");
   const isGraphQL = pathname.startsWith("/graphql");
 
+  // @ts-expect-errors
```

Comment:
Can this TS error be solved with the same approach used in `types.ts` (see also my comment there)?

Reply:
I'll have to look into it, but I'm not sure I found a way to fix it.

It just caused more errors without it

Reply:
I managed to remove some more, including this one. And added an explanation comment